### ISSUE TITLE
IoTHub Fallback Route should be enabled by default

### ISSUE THREAD

**Issue by KamalAman** [terraform-providers/terraform-provider-azurerm#7571]
IoTHub Fallback Route should be abled by default. This seems to be the developers original intension in [this PR #2764](https://github.com/terraform-providers/terraform-provider-azurerm/pull/2764) that introduces the fallback route

### Terraform (and AzureRM Provider) Version
2.13.0

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "Canada Central"
}

resource "azurerm_iothub" "example" {
  name                = "Example-IoTHub"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location

  sku {
    name     = "S1"
    capacity = "1"
  }
}
```

### Expected Behavior

The fallback route is not enabled

### Actual Behavior
Itt should be enabled by default

### Steps to Reproduce
1. `terraform apply`

**Comment by lrxtom2** [comment]
Thanks for opening the issue! @KamalAman 
We noticed that the `Fallback Route` is enabled by default in portal. But at this time we will not implement this change in Terraform. As a workaround please add `fallback_route` block in your tf code like below or to use resource of [`iothub_fallback_route`](https://www.terraform.io/docs/providers/azurerm/r/iothub_fallback_route.html) what you mentioned instead .

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "Canada Central"
}

resource "azurerm_iothub" "example" {
  name                = "Example-IoTHub"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location

  sku {
    name     = "S1"
    capacity = "1"
  }

  fallback_route {
   source         = "DeviceMessages"
    endpoint_names = ["events"]
    enabled        = true
  }
}
```

**Comment by KamalAman** [comment]
Thank for the information @lrxtom2. Whats the reason why you won't change the default? 

In Addition, the default  setting for the fallback route is not explicitly defined in the documentation [fallback_route](https://www.terraform.io/docs/providers/azurerm/r/iothub.html#fallback_route). It might be worth while to update the documentation to explicitly state that the fallback route is not enabled by default.

**Comment by ms-vincent** [comment]
Agreed that at the very least, the documentation should be updated to state that the fallback route is not enabled. If configured incorrectly, disabling of the fallback route can cause issues which will be hard to troubleshoot for the unaware.

**Comment by katbyte** [comment]
Hi @KamalAman - thanks for opening this issue. While we can't make this change in 2.0 as it could break existing users, we can make it in 3.0 🙂 - until then we can update the documentation to reflect the current behaviour.

**Comment by machteldbogels** [comment]
@lrxtom2 question related to this issue - when I specify "events" for endpoint_names during creation of the `iothub_fallback_route` resource, I receive the error that the reserved "events" endpoint cannot be used for custom endpoints. Is there another way to ensure this separate fallback route resource will use the built-in endpoint?
